packages/mobile/src/backup/BackupComplete.tsx

```diff
@@ -1,44 +1,32 @@
 import Checkmark from '@celo/react-components/icons/Checkmark'
 import fontStyles from '@celo/react-components/styles/fonts'
 import { StackScreenProps } from '@react-navigation/stack'
-import * as React from 'react'
-import { WithTranslation } from 'react-i18next'
+import React, { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
 import { StyleSheet, Text, View } from 'react-native'
 import { SafeAreaView } from 'react-native-safe-area-context'
-import { connect } from 'react-redux'
 import { OnboardingEvents } from 'src/analytics/Events'
 import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
-import { withTranslation } from 'src/i18n'
 import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { StackParamList } from 'src/navigator/types'
-import { RootState } from 'src/redux/reducers'
-
-interface StateProps {
-  backupCompleted: boolean
-}
-
-type Props = StateProps & WithTranslation & StackScreenProps<StackParamList, Screens.BackupComplete>
-
-const mapStateToProps = (state: RootState): StateProps => {
-  return {
-    backupCompleted: state.account.backupCompleted,
-  }
-}
+import useSelector from 'src/redux/useSelector'
 
 /**
  * Component shown to the user upon completion of the Recovery Phrase setup flow. Informs the user that
  * they've successfully completed the backup process and automatically returns them to where they
  * came from.
  */
-class BackupComplete extends React.Component<Props> {
-  static navigationOptions = { header: null }
 
-  componentDidMount() {
-    // Show success check for a while before leaving screen
-    const { backupCompleted } = this.props
+type Props = StackScreenProps<StackParamList, Screens.BackupComplete>
+
+function BackupComplete({ route }: Props) {
+  const navigatedFromSettings = route.params?.navigatedFromSettings ?? false
+  const backupCompleted = useSelector((state) => state.account.backupCompleted)
```

Comment:
nit: should we create a selector here, to keep consistency? i'm always unsure about the benefit of using selectors vs doing the selection directly like this, i think for simple selectors like this there isn't much difference but there are already some simple selectors in the `src/account/selectors.ts` file so would be nice to keep the pattern?

Reply:
Should be fixed in https://github.com/valora-inc/wallet/pull/1676/commits/ddc035bd8314453a01fde0f89275674f373f3a86!